apps/web-giddh/src/app/settings/integration/payment/icici/payor-create-edit/payor-create-edit.component.ts

```diff
@@ -163,6 +163,12 @@ export class PayorCreateEditComponent implements OnInit, OnDestroy {
                 this.accountUserForm.get('duration')?.patchValue(UNLIMITED_LIMIT);
             }
 
+            if((!this.accountUserForm.get('duration')?.value || this.accountUserForm.get('duration')?.value === UNLIMITED_LIMIT) && this.accountUserForm.get('maxAmount')?.value) {
```

Comment:
Consider moving this code block to a method since it is repeated in the same file (line 105 & 166)